src/codegen/ast_interpreter.cpp

```diff
@@ -1115,38 +1152,41 @@ Box* astInterpretFunction(CompiledFunction* cf, int nargs, Box* closure, Box* ge
     return v.o ? v.o : None;
 }
 
-Box* astInterpretFunctionEval(CompiledFunction* cf, BoxedDict* locals) {
+Box* astInterpretFunctionEval(CompiledFunction* cf, Box* boxedLocals) {
     ++cf->times_called;
 
     ASTInterpreter interpreter(cf);
-    for (const auto& p : locals->d) {
-        assert(p.first->cls == str_cls);
-        auto name = static_cast<BoxedString*>(p.first)->s;
-        InternedString interned = cf->clfunc->source->getInternedStrings().get(name);
-        interpreter.addSymbol(interned, p.second, false);
-    }
-
     interpreter.initArguments(0, NULL, NULL, NULL, NULL, NULL, NULL);
+    RELEASE_ASSERT(boxedLocals->cls == dict_cls, "we don't support non-dicts here yet");
+    interpreter.setBoxedLocals(boxedLocals);
     Value v = ASTInterpreter::execute(interpreter);
 
     return v.o ? v.o : None;
 }
 
 Box* astInterpretFrom(CompiledFunction* cf, AST_expr* after_expr, AST_stmt* enclosing_stmt, Box* expr_val,
-                      BoxedDict* locals) {
+                      BoxedDict* stackLocals) {
```

Comment:
Maybe we should pass the entire FrameInfo object?  Feels like we need to propagate the exc_info as well.